builder/infer_bounds.cc

```diff
@@ -76,13 +76,22 @@ std::vector<dim_expr> recursive_substitute(
 // producers to the required region.
 class bounds_inferrer : public node_mutator {
 public:
+  node_context& ctx;
+
   symbol_map<box_expr> infer;
   symbol_map<box_expr> crops;
 
+  // A map of buffers that should be substituted in for consumers.
+  symbol_map<symbol_id> consumer_subs;
+
+  bounds_inferrer(node_context& ctx) : ctx(ctx) {}
+
   void visit(const allocate* op) override {
+    symbol_id uncropped = ctx.insert_unique(ctx.name(op->sym) + ".uncropped");
```

Comment:
There's still a big difference, in that we aren't using the name as meaningful information. We can change this name and nothing will break. Giving it a meaningful name just makes the printed nodes easier to read and understand.

Reply:
IMO the code smell to avoid is ever parsing one of these names, or comparing it to a constructed name. E.g. never do `if (op->name == my_func + ".buffer") ...`

Reply:
And I think the alternative, for the case I mention above, is to have a dedicated IR node that means the buffer associated with the func of the given name.

Reply:
Ah, so this is just "make a probably-unique name for the node", gotcha